### PR TITLE
Allow to set API icons in @Api

### PR DESCRIPTION
**Pull request by clementdenis**
Use new Google logo for default icons (aligned with Google APIs)

### CODE REVIEW

**Review by funkomatic — APPROVED** [review]

File: endpoints-framework/src/main/java/com/google/api/server/spi/config/jsonwriter/JsonConfigWriter.java [hunk 1]
```diff
@@ -171,6 +171,12 @@ private void convertApi(ObjectNode root, ApiConfig config) {
     if (config.getDocumentationLink() != null) {
       root.put("documentation", config.getDocumentationLink());
     }
+    if (config.getIconX16() != null || config.getIconX32() != null) {
+      ObjectNode icons = objectMapper.createObjectNode();
+      icons.put("x16", config.getIconX16());
+      icons.put("x32", config.getIconX32());
```
Comment (alarribeau) [inline endpoints-framework/src/main/java/com/google/api/server/spi/config/jsonwriter/JsonConfigWriter.java]: Shouldn't we handle the case were just one of them has been provided ?
```
if (config.getIconX16() != null) {
  icons.put("x16", config.getIconX16());
}
if (config.getIconX32() != null) {
  icons.put("x32", config.getIconX32());
}
```

File: endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java [hunk 2]
```diff
@@ -422,8 +428,8 @@ private DirectoryList generateDirectory(Map<ApiKey, RestDescription> discoveryDo
           .setDiscoveryLink("." + relativePath)
           .setDiscoveryRestUrl(context.getApiRoot() + "/discovery/v1" + relativePath)
           .setIcons(new Icons()
-                .setX16("https://www.gstatic.com/images/branding/product/1x/googleg_16dp.png")
-                .setX32("https://www.gstatic.com/images/branding/product/1x/googleg_32dp.png"))
+              .setX16(doc.getIcons().getX16())
+              .setX32(doc.getIcons().getX32()))
```
Comment (alarribeau) [inline endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java:432]: A fallback to the (new?) google default icons would make this feature backward compatible.

**Review by alarribeau — APPROVED** [review]
Approved with some optional remarks

File: endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java (hunk 2 above)
Comment (clementdenis) [reply endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java:432]: It is backward compatible: the default icons are set in REST_SKELETON, which is cloned to generate RestDescription instances in DicoveryGenerator (the previous code was actually useless ...).

File: endpoints-framework/src/main/java/com/google/api/server/spi/config/jsonwriter/JsonConfigWriter.java (hunk 1 above)
Comment (clementdenis) [reply endpoints-framework/src/main/java/com/google/api/server/spi/config/jsonwriter/JsonConfigWriter.java]: I updated the code to require both icon sizes to be set (don't want inconsistent display).
This is not a blocking error however to set only one version though.

File: endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java (hunk 2 above)
Comment (alarribeau) [reply endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java:432]: Ok cool, I missed it.

**Review by alarribeau — APPROVED** [review]

**Pull request closed without merging**